### ISSUE TITLE
Password reset should not confirm or deny existence of account

### ISSUE THREAD

**Issue by amplifi** [Cadasta/cadasta-platform#1272]
### Steps to reproduce the error

Attempt a password reset for an email address that does not correspond to a platform user account.

### Actual behavior

The error message indicates there's no user account for that email address. This of course means that if the reset password submission is approved, a valid account exists for that email address.

<img width="500" alt="screen shot 2017-03-10 at 6 15 19 pm" src="https://cloud.githubusercontent.com/assets/4269749/23791458/0543c86a-05be-11e7-8b26-44fecaae0b32.png">

### Expected behavior

The reset password process shouldn't confirm or deny whether an account exists for a given email address.

**Comment by aklife97** [comment]
Hi, I would like to take up this issue.
Django allauth does not by default provide this functionality as we can see [here](https://github.com/pennersr/django-allauth/issues/1307)
So as a solution
I can create `ResetPasswordForm` class and have  a custom clean_email method in it.
Now from inside the function, I can check simply get the email and put the user in self.users
Now save would be run (even when there is an error)but as there is no self.users, nothing would happen.

**Comment by wonderchook** [comment]
@amplifi should we also not confirm that an email has been sent?

**Comment by aklife97** [comment]
@wonderchook I just sent a PR. I have also added a 2 tests to check if mail is sent or not. This should suffice to confirm whether email has been sent.

**Comment by amplifi** [comment]
@wonderchook I usually see messages similar to "Your email address has been submitted. If it matches an account on the Cadasta Platform, you will receive an email shortly with instructions for resetting your password." etc

**Comment by wonderchook** [comment]
@aklife97 this issue was not marked new contributor friendly nor had we decided exactly how to fix it yet. Can you review this before starting on another issue: https://github.com/Cadasta/cadasta-project-ideas/wiki/GSoC-Outreachy-Issue-Contribution-Process

**Comment by aklife97** [comment]
@wonderchook Okay, Sorry. Would keep it in mind.

**Comment by dpalomino** [comment]
> @wonderchook I usually see messages similar to "Your email address has been submitted. If it matches an account on the Cadasta Platform, you will receive an email shortly with instructions for resetting your password." etc

I think that would be sufficient...
